tests/base/src/test/java/org/wildfly/security/audit/SyslogAuditEndpointTest.java

```diff
@@ -0,0 +1,301 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Test cases to test {@link SyslogAuditEndpoint}
+ *
+ * @author <a href="mailto:jucook@redhat.com">Justin Cook</a>
+ */
+public class SyslogAuditEndpointTest {
+    private static final String HOST_NAME = "localhost";
+    private static final String BAD_HOST_NAME = "0.0.0.1";
+    private static final String RFC3164_LOG_MESSAGE = "testing log message RFC3164";
+    private static final String RFC5424_LOG_MESSAGE = "testing log message RFC5424";
+    private static final int UDP_PORT = 10837;
+    private static final int TCP_PORT = 10857;
+    private static final int RECONNECT_NUMBER = 10;
+    private static final int BAD_RECONNECT_NUMBER = -2;
+    private static final int RECONNECT_TIMEOUT = 10;
+    private static SocketFactory socketFactory = null;
+    private static SimpleSyslogServer udpServer = null;
+    private static SimpleSyslogServer tcpServer = null;
+
+    /**
+     * Gets the server address and makes the socket factory before starting up the server
+     */
+    @BeforeClass
+    public static void initializeServer() throws Exception {
+        socketFactory = SocketFactory.getDefault();
+        udpServer = SimpleSyslogServer.createUdp(UDP_PORT);
+        tcpServer = SimpleSyslogServer.createTcp(TCP_PORT, false);
+    }
+
+    /**
+     * Safely shutdowns the server
+     */
+    @AfterClass
+    public static void shutdownServer() {
+        udpServer.close();
+        tcpServer.close();
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 3164 format and critical priority
+     */
+    @Test
+    public void testGenericRfc3164SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC3164);
+        endpoint.accept(EventPriority.CRITICAL, RFC3164_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC3164_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 5424 format and critical priority
+     */
+    @Test
+    public void testGenericRfc5424SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC5424);
+        endpoint.accept(EventPriority.CRITICAL, RFC5424_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC5424_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the endpoint rejects a bad integer
+     */
+    @Test
+    public void testBadIntegerReconnectAttempts() throws Exception {
+        try {
+            // Since the endpoint gives a message upon creation there is no need to call .accept()
+            AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(BAD_RECONNECT_NUMBER);
+        } catch (IllegalArgumentException e) {
+            assertEquals(e.getMessage(), "ELY11008: The reconnect attempts value of " + BAD_RECONNECT_NUMBER + " is invalid. Please use an integer value >= -1.");
```

Comment:
I think it would be better to remove "ELY11008:" and just check the message contains specific text.

Reply:
Fixed